builtin/providers/aws/resource_aws_db_instance.go

```diff
@@ -916,13 +916,37 @@ func resourceAwsDbInstanceUpdate(d *schema.ResourceData, meta interface{}) error
 		requestUpdate = true
 	}
 
+	if d.HasChange("port") {
+		d.SetPartial("port")
+		req.DBPortNumber = aws.Int64(int64(d.Get("port").(int)))
+		requestUpdate = true
+	}
+
 	log.Printf("[DEBUG] Send DB Instance Modification request: %t", requestUpdate)
 	if requestUpdate {
 		log.Printf("[DEBUG] DB Instance Modification request: %s", req)
 		_, err := conn.ModifyDBInstance(req)
 		if err != nil {
 			return fmt.Errorf("Error modifying DB Instance %s: %s", d.Id(), err)
 		}
+
+		log.Println("[INFO] Waiting for DB Instance to be available")
```

Comment:
> The state of the database goes back to available after it has finished updating

I think this is true for some attributes, like `port`, which happen to change quickly. I'm not sure what others there that behave like this, but things like `allocated_storage` do not. We still see a plan/diff on follow up if we're not using the `apply_immediately` flag. 

That said, it prevents unnecessary plan/diffs for things like `port`, so we'll go with it 👍